app/services/workers/reductions-worker.js

```diff
@@ -0,0 +1,72 @@
+const getAllOpenReductions = require('../data/get-all-open-reductions')
+const updateReductionStatusByIds = require('../data/update-reduction-status-by-ids')
+const reductionStatus = require('../../constants/reduction-status')
+const createNewTasks = require('../data/create-tasks')
+const taskType = require('../../constants/task-type')
+const taskStatus = require('../../constants/task-status')
+const Task = require('../domain/task')
+const submittingAgent = require('../../constants/task-submitting-agent')
+const logger = require('../log')
+
+module.exports.execute = function (task) {
+  var idsMap = new Map()
+  idsMap.set(reductionStatus.ACTIVE, [])
+  idsMap.set(reductionStatus.SCHEDULED, [])
+  idsMap.set(reductionStatus.DELETED, [])
+  idsMap.set(reductionStatus.ARCHIVED, [])
+
+  logger.info('Retrieving open reductions')
+  return getAllOpenReductions()
+    .then(function (reductions) {
+      reductions.forEach(function (reduction) {
+        status = getReducitonStatus(reduction)
```

Comment:
Check does the status change before adding to the list of ids that need processed?

Reply:
Added, also added unit test